### PR TITLE
fix: make text non selectable if inside a button [web]

### PR DESCRIPTION
**Pull request by RichardLindhout**
### Summary
Provide same behaviour across platforms since text is non-selectable by default on other platforms.
Fixes: https://github.com/callstack/react-native-paper/issues/2258
94059145-50e76680-fde2-11ea-82d9-8c01f65f6415

### CODE REVIEW

**Review by satya164 — APPROVED** [review]

**Comment by Trancever** [comment]
@RichardLindhout Thanks for the contributions! Could you resolve the conflicts and make sure lint and tests are passing? Once it's done we would like to merge the PR.

**Comment by RichardLindhout** [comment]
Yes, will try to look at it this week a little bit short on time a.t.m.

**Comment by RichardLindhout** [comment]
@Trancever done

**Comment by RichardLindhout** [comment]
@Trancever We still need it inside the react-native-vector package but I don't think the maintainer will add this anytime soon based on all the PR's which are open.

I'll add it here too!

File: flow-typed/npm/babel-preset-react-native_vx.x.x.js [hunk 1]
```diff
@@ -60,14 +60,20 @@ declare module 'babel-preset-react-native/index.js' {
   declare module.exports: $Exports<'babel-preset-react-native'>;
 }
 declare module 'babel-preset-react-native/lib/resolvePlugins.js' {
-  declare module.exports: $Exports<'babel-preset-react-native/lib/resolvePlugins'>;
```
Comment (Trancever) [inline flow-typed/npm/babel-preset-react-native_vx.x.x.js:63]: Why did you make all these changes in flow-types directory? You must have run prettier on it by mistake.

Btw, I don't think we need flow type definitions anymore, but let's remove it in separate PR.

Comment (Trancever) [reply flow-typed/npm/babel-preset-react-native_vx.x.x.js:63]: Please revert all not related changes if possible 🙏

File: package.json [hunk 2]
```diff
@@ -40,7 +40,8 @@
     "release": "release-it",
     "bootstrap": "yarn --cwd example && yarn --cwd docs && yarn",
     "docs": "yarn --cwd docs",
-    "example": "yarn --cwd example"
+    "example": "yarn --cwd example",
+    "prettier": "./node_modules/.bin/prettier --write './src/**/*.{tsx,ts,js,css,html}'"
```
Comment (Trancever) [inline package.json:44]: I don't think we need that script, it should be handled by the code editor with on save action.

Anyway, It should run the prettier on src and example directories only and if we really want that then let's do it in a separate PR, please.

**Review by Trancever — CHANGES_REQUESTED** [review]

File: flow-typed/npm/babel-preset-react-native_vx.x.x.js (hunk 1 above)
Comment (RichardLindhout) [reply flow-typed/npm/babel-preset-react-native_vx.x.x.js:63]: Will do, did indeed run prettier outside of the src

**Comment by Trancever** [comment]
@RichardLindhout What's the status of it? Let me know if you don't have time to work on this one so someone from the core team will finish it. Cheers!

**Comment by RichardLindhout** [comment]
Thanks for the feedback! It would be nice if the core-team could cherry-picking the commits since I had a hard time to fix this the last time since I don't 'get' how that works

**Comment by RichardLindhout** [comment]
Hey @Trancever I have fixed some new conflicts, could we maybe merge this soon?

**Comment by Trancever** [comment]
@RichardLindhout I cherry-picked only the relevant commit and it's already on a master!

**Pull request closed without merging**